api/app/usecases/users/create.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require './app/domain/user'
+require './app/domain/user_repository'
+
+module Usecases
+  module Users
+    class InvalidUser < StandardError; end
+
+    class Create
+      attr_accessor :user_repository, :user, :params
+
+      def initialize(params:, repository_adapter:, user: Domain::User)
+        @params = params
+        @user = user
+        @user_repository = Domain::UserRepository.new(repository: repository_adapter)
```

Comment:
Isso aqui deveria estar como default no construtor talvez?

Reply:
Eu pensei sem deixar dessa forma para meio que deixar o use-case "amarrado" com o Domain e quem for chama-lo somente passar o adapter como param.